### DISCUSSION TITLE
Issue with installation of SeisSol/preprocessing/meshing/cube_c

### DISCUSSION THREAD

**Discussion by vikaskurapati** [SeisSol/SeisSol#865] [Q&A] [answered]
Hi, 

I could not find a proper documentation and am unable to install the submodule cube_c. I am not sure if I am looking at the right place for it. Nevertheless, I tried building it normally using the CMakeLists.txt provided and it gives the following error:

/usr/bin/ld: /home/vikaskurapati/lib/libhdf5.a(H5Zdeflate.o): undefined reference to symbol 'inflateEnd'
/usr/bin/ld: /lib/x86_64-linux-gnu/libz.so.1: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/cubeGenerator.dir/build.make:103: cubeGenerator] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/cubeGenerator.dir/all] Error 2
make: *** [Makefile:91: all] Error 2

Am I missing some necessary flags to get this done?

**Comment by davschneller** [comment, marked as answer]
Hi, you can try to add `-lz` to your linker flags in CMake, i.e. link to libz explicitly.

(alternatively, you could try to add `-Wl,--copy-dt-needed-entries` (cf. e.g. here https://linuxpip.org/how-to-fix-dso-missing-from-command-line/ ))

**Reply by vikaskurapati** [reply]
Thank you, it works! :)
